src/box/vinyl.c

```diff
@@ -4396,8 +4396,18 @@ vy_deferred_delete_on_replace(struct trigger *trigger, void *event)
 	 * process the batch (we can't yield between statements).
 	 */
 	struct vy_env *env = vy_env(space->engine);
-	if (is_first_statement)
+	if (is_first_statement) {
 		vy_quota_wait(&env->quota, VY_QUOTA_CONSUMER_COMPACTION);
+		/*
+		 * The space could have been dropped while we were
+		 * waiting for quota.
+		 */
+		space = space_cache_find(space_id);
```

Comment:
What if another space with the same ID is created while we are waiting?

Reply:
I think we should finally add `unique_id` member to struct space. Having space_id and unique_id we can always definitely determine that the space is still the same.

Reply:
Right, I missed that. Thanks for catching this. Fixed by checking `vy_lsm->is_dropped` after the yield. Also added a test checking this case.